### DISCUSSION TITLE
How to run an action that will be run by `repository_dispatch` event?

### DISCUSSION THREAD

**Discussion by FeryET** [nektos/act#2225] [Q&A]
I am trying to run an action that should be run when a `repository_dispatch` event is issued in another workflow.

My current `event.json` is like this:

```json
{
        "event_type": "deploy.staging",
        "client_payload": {
                "service_name": "frontend",
                "service_version": "v0.1.9"
        }

}
```

And I am submitting my workflow via this command:

```sh
$ act 'repository_dispatch' -e event.json --secret TOKEN="$secret_token"  
```

But the variables related to payload are empty in the above command. Am I running this correctly?

**Comment by excalq** [comment]
This is how I test `repository_dispatch` events with act, keeping the JSON value inline, though referencing a `.json` file is fine. The key seems to the `github.events` JSON having `action` and `client_payload` keys.

```
act repository_dispatch --verbose --workflows .github/workflows/web-ui-deployment.yaml --secret GITHUB_TOKEN="$(gh auth token)" \
  --eventpath <(echo '{"action": "staging-deployment-web-ui", "client_payload": {"tag": "v0.1.2"}}') \
  --var AWS_REGION=$AWS_REGION --var APP_ENVIRONMENT=$APP_ENVIRONMENT
```
